tensorflow-tools/src/main/java/org/tensorflow/tools/buffer/impl/raw/UnsafeMemoryHandle.java

```diff
@@ -138,39 +142,47 @@ int arrayOffset(Class<?> arrayClass) {
   }
 
   ByteBuffer toArrayByteBuffer() {
-    return ByteBuffer.wrap((byte[])object, (int)byteOffset - UnsafeReference.UNSAFE.arrayBaseOffset(byte[].class), (int)size());
+    return ByteBuffer.wrap((byte[])object, (int) byteOffset - UnsafeReference.UNSAFE.arrayBaseOffset(byte[].class), (int)size);
   }
 
   ShortBuffer toArrayShortBuffer() {
-    return ShortBuffer.wrap((short[])object, (int)((byteOffset - UnsafeReference.UNSAFE.arrayBaseOffset(short[].class)) / scale), (int)size());
+    return ShortBuffer.wrap((short[])object, (int)((byteOffset - UnsafeReference.UNSAFE.arrayBaseOffset(short[].class)) / scale), (int)size);
   }
 
   IntBuffer toArrayIntBuffer() {
-    return IntBuffer.wrap((int[])object, (int)((byteOffset - UnsafeReference.UNSAFE.arrayBaseOffset(int[].class)) / scale), (int)size());
+    return IntBuffer.wrap((int[])object, (int)((byteOffset - UnsafeReference.UNSAFE.arrayBaseOffset(int[].class)) / scale), (int)size);
   }
 
   LongBuffer toArrayLongBuffer() {
-    return LongBuffer.wrap((long[])object, (int)((byteOffset - UnsafeReference.UNSAFE.arrayBaseOffset(long[].class)) / scale), (int)size());
+    return LongBuffer.wrap((long[])object, (int)((byteOffset - UnsafeReference.UNSAFE.arrayBaseOffset(long[].class)) / scale), (int)size);
   }
 
   FloatBuffer toArrayFloatBuffer() {
-    return FloatBuffer.wrap((float[])object, (int)((byteOffset - UnsafeReference.UNSAFE.arrayBaseOffset(float[].class)) / scale), (int)size());
+    return FloatBuffer.wrap((float[])object, (int)((byteOffset - UnsafeReference.UNSAFE.arrayBaseOffset(float[].class)) / scale), (int)size);
   }
 
   DoubleBuffer toArrayDoubleBuffer() {
-    return DoubleBuffer.wrap((double[])object, (int)((byteOffset - UnsafeReference.UNSAFE.arrayBaseOffset(double[].class)) / scale), (int)size());
+    return DoubleBuffer.wrap((double[])object, (int)((byteOffset - UnsafeReference.UNSAFE.arrayBaseOffset(double[].class)) / scale), (int)size);
   }
 
   final Object object;
-  final long byteOffset;
+  final long baseOffset;
+  long byteOffset;
```

Comment:
This is the crucial change right? Making this class no longer completely immutable so it can walk down an allocated block of memory.

Reply:
It is... but the whole point of this PR is to have somewhere a mutable object that can be recycled. Note that this class is package-scoped, so nothing outside this internal package can change the value of it other that the endpoints we expose (which is `window.slide()` in this case).

Still, do you think that might be worth having two version of this object, one mutable and one not?

Reply:
It's fine as is, I was just checking that I'd understood the PR properly. I don't think it will make too much difference performance wise having a mutable field, unless it's on a JVM with more aggressive escape analysis like Graal (where it might be able to elide the allocation, but it might be able to do that for the mutable version too).